plugins/module_utils/common/api/onemanage/endpoints.py

```diff
@@ -0,0 +1,1647 @@
+    """
+
+    vrf_names: Optional[str] = Field(None, min_length=1, description="Comma-separated VRF names")
+
+    def to_query_string(self) -> str:
+        """Build query string with vrf-names parameter."""
+        if self.vrf_names:
+            return f"vrf-names={self.vrf_names}"
+        return ""
+
+
+class EpOneManageFabricConfigDeploy(BaseModel):
+    """
+    ## Fabric Config-Deploy Endpoint (OneManage)
+
+    ### Description
+    Endpoint to deploy the configuration for a specific multi-cluster fabric.
+
+    ### Path
+
+    - /appcenter/cisco/ndfc/api/v1/onemanage/fabrics/{fabricName}/config-deploy
+
+    ### Verb
+
+    - POST
+
+    ### Usage
+    ```python
+    request = EpOneManageFabricConfigDeploy()
+    request.fabric_name = "MyFabric"
+    request.query_params.force_show_run = "true"
+    request.query_params.incl_all_msd_switches = "false"
+
+    path = request.path
+    verb = request.verb
+    ```
+    """
+
+    class_name: Optional[str] = Field(default="EpOneManageFabricConfigDeploy", description="Class name for backward compatibility")
+    fabric_name: Optional[str] = Field(None, min_length=1, description="Fabric name")
+    query_params: FabricConfigDeployQueryParams = Field(default_factory=FabricConfigDeployQueryParams)
+
+    def __init__(self, **data):
+        """Initialize with default query parameter objects."""
+        super().__init__(**data)
+        if not isinstance(self.query_params, FabricConfigDeployQueryParams):
+            self.query_params = FabricConfigDeployQueryParams()
+
+    @property
+    def path(self) -> str:
+        """
+        Build the endpoint path with query parameters.
+
+        ### Raises
+        - ValueError: If fabric_name is not set
+
+        ### Returns
+        - Complete endpoint path string with query parameters
+        """
+        if self.fabric_name is None:
+            raise ValueError("fabric_name must be set before accessing path")
+
+        base_path = BasePath.onemanage_fabrics(self.fabric_name, "config-deploy")
+
+        query_string = self.query_params.to_query_string()
+        if query_string:
+            return f"{base_path}?{query_string}"
+        return base_path
+
+    @property
+    def verb(self) -> Literal["POST"]:
+        """Return the HTTP verb for this endpoint."""
+        return "POST"
+
+
+class EpOneManageFabricConfigDeploySwitch(BaseModel):
+    """
+    ## Fabric Config-Deploy Switch Endpoint (OneManage)
+
+    ### Description
+    Endpoint to deploy the configuration for a specific switch in a multi-cluster fabric.
+
+    ### Path
+
+    - /appcenter/cisco/ndfc/api/v1/onemanage/fabrics/{fabricName}/config-deploy/{switchSN}
+
+    ### Verb
+
+    - POST
+
+    ### Usage
+    ```python
+    request = EpOneManageFabricConfigDeploySwitch()
+    request.fabric_name = "MyFabric"
+    request.switch_sn = "92RZ2OMQCNC"
+    request.query_params.force_show_run = "true"
```

Comment:
I might be missing something here but is there a way to use this endpoint class but NOT have it append the query_pararms?